### DISCUSSION TITLE
First time using the template resulted in failure

### DISCUSSION THREAD

**Discussion by jwoverland** [CloudBoltSoftware/cb-applet#23] [Q&A] [answered]
I was able to build the CB Applet however I had to run the post-build separately as it fails to run as part of the build.

I uploaded the zip file to UI Manager and upon switching to the CUI I'm immediately presented with an error notification reading "Failed to load applets".

We are running CMP v2023.4.1 and recently migrated it from MySQL to Postgres.

**Comment by orvisevans** [comment]
Thanks for the feedback on your issues! I'll take them one at a time.

## Separate post-build

The post-build step ran for me without an error on a clean project using these commands:
```bash
gh repo clone CloudBoltSoftware/cb-applet
cd cb-applet
npm install
npm run build
```
I can't reproduce the issue you're having here, so I'm guessing our systems are subtly different. I'm running Node 18.16 on a Mac. What is your OS and Node version?

## Failed to load applets Error

I was able to reproduce this after running the above commands on a similar version of CloudBolt. The root cause was the post-build command didn't bundle the applet code into the `.zip`, so when CloudBolt tried to serve the applet, there was nothing there to serve.

This worked for me as a workaround.
1. Delete the uploaded Applet
2. re-run `npm run build`
3. Upload the new Applet

For some reason, running `build` a second time bundled the code in correctly. You can double-check this after building by looking in the zip, then in the internal zip, and seeing if there's a `static` folder with a `.js` script inside. I'll see if I can track down why this is happening.

**Comment by orvisevans** [comment, marked as answer]
I've pushed a fix to the load applets Error. It shouldn't be a problem any more.

**Reply by jwoverland** [reply]
Your commit resolved the issue of failing to load the applet in the CUI which is great and will allow me to start some testing.

I still have to run the post-build as a separate operation after the build (not the end of the world).  I am running node v17.4.0 on my work computer and v18.17.0 on my personal computer and both produce the same error as follows:

> vite v4.3.9 building for production...
✓ 432 modules transformed.
(post-build-plugin plugin) Running post build script: "npm run post-build"
computing gzip size (0)...Error: spawn npm ENOENT
    at Process.ChildProcess._handle.onexit (node:internal/child_process:283:19)
    at onErrorNT (node:internal/child_process:476:16)
    at processTicksAndRejections (node:internal/process/task_queues:83:21) {
  errno: -4058,
  code: 'ENOENT',
  syscall: 'spawn npm',
  path: 'npm',
  spawnargs: [ 'run', 'post-build' ]
}
(post-build-plugin plugin) post build script "npm run post-build" exited with code -4058
